spec/patterns/shorewall_spec.rb

```diff
@@ -2,89 +2,101 @@
+                                "length"=>141,
+                                "tos"=>"00", "precedence_bits"=>"00",
+                                "ttl"=>63,
+                                "id"=>"55251"
+                            },
+                            "network"=>{"transport"=>"UDP"},
+                            "source"=>{"ip"=>"1.2.3.4", "port"=>5353},
+                            "destination"=>{"ip"=>"192.168.0.10", "port"=>5335}
+                        )
+      else
+        should include("nf_host" => "myHost")
+        should include("nf_action1" => "FORWARD")
+        should include("nf_action2" => "REJECT")
+        should include("nf_in_interface" => "eth2")
+        should include("nf_out_interface" => "eth2")
+        should include("nf_src_ip" => "1.2.3.4")
+        should include("nf_dst_ip" => "192.168.0.10")
+        should include("nf_len" => "141")
+        should include("nf_tos" => "0x00")
+        should include("nf_prec" => "0x00")
+        should include("nf_ttl" => "63")
+        should include("nf_id" => "55251")
+        should include("nf_protocol" => "UDP")
+        should include("nf_src_port" => "5353")
+        should include("nf_dst_port" => "5335")
+      end
+    end
 
-    it { should include("nf_dst_port" => "5353") }
   end
 
   context "parsing a message without OUT interface" do
 
-    let(:value) { "May 28 17:31:07 myHost kernel: [3125121.106700] Shorewall:net2fw:DROP:IN=eth1 OUT= MAC=00:02:b3:c7:2f:77:38:72:c0:6e:92:9c:08:00 SRC=1.2.3.4 DST=1.2.3.4 LEN=60 TOS=0x00 PREC=0x00 TTL=49 ID=6480 DF PROTO=TCP SPT=59088 DPT=8080 WINDOW=2920 RES=0x00 SYN URGP=0" }
-
-    subject     { grok_match(pattern, value) }
-
-    it { should include("timestamp" => "May 28 17:31:07") }
-
-    it { should include("nf_host" => "myHost") }
-
-    it { should include("nf_action1" => "net2fw") }
-
-    it { should include("nf_action2" => "DROP") }
-
-    it { should include("nf_in_interface" => "eth1") }
-
-    it { expect(subject["nf_out_interface"]).to be_nil }
-
-    it { should include("nf_dst_mac" => "00:02:b3:c7:2f:77") }
-
-    it { should include("nf_src_mac" => "38:72:c0:6e:92:9c") }
-
-    it { should include("nf_src_ip" => "1.2.3.4") }
-
-    it { should include("nf_dst_ip" => "1.2.3.4") }
-
-    it { should include("nf_len" => "60") }
-
-    it { should include("nf_tos" => "0x00") }
-
-    it { should include("nf_prec" => "0x00") }
-
-    it { should include("nf_ttl" => "49") }
-
-    it { should include("nf_id" => "6480") }
-
-    it { should include("nf_protocol" => "TCP") }
 
-    it { should include("nf_src_port" => "59088") }
+    let(:message) do
+      "May 28 17:31:07 server Shorewall:net2fw:DROP:" +
+      "IN=eth1 OUT= MAC=00:02:b3:c7:2f:77:38:72:c0:6e:92:9c:08:00 SRC=127.0.0.1 DST=1.2.3.4 LEN=60 TOS=0x00 PREC=0x00 TTL=49 ID=6480 DF PROTO=TCP SPT=59088 DPT=8080 WINDOW=2920 RES=0x00 SYN URGP=0"
+    end
+
+    it 'matches' do
+      should include("timestamp" => "May 28 17:31:07")
```

Comment:
nit: I really don't like the `should` short-hand with implicit subject. While it makes the `it { should ` ... ` }` one-liners more "readable", I think it does the opposite for long-form blocks.

I would much prefer an explicit subject:

~~~ ruby
        subject(:the_event) { grok_match(pattern, value) }
~~~

And the `expect` syntax:

~~~
  it 'matches' do
    expect(the_event).to include("timestamp" => "May 28 17:31:07")
    # ...
  end
~~~

Reply:
yeah, :+1: also hate the should syntax.
these specs did get boring so I ended up :scissors: based on legacy, instead of rewriting everything to expect.

Reply:
might address some of these refactorings in the second part, so that we can get a review for the whole pattern set.